src/android/app/src/main/jni/vr/vr_main.cpp

```diff
@@ -472,9 +486,95 @@ class VRApp {
             OXR(xrCreateReferenceSpace(gOpenXr->mSession, &sci, &gOpenXr->mHeadSpace));
         }
 
+        gOpenXr->headLocation = {XR_TYPE_SPACE_LOCATION};
+        OXR(xrLocateSpace(gOpenXr->mViewSpace, gOpenXr->mHeadSpace, frameState.predictedDisplayTime, &gOpenXr->headLocation));
+
         mInputStateFrame.SyncHandPoses(gOpenXr->mSession, mInputStateStatic, gOpenXr->mLocalSpace,
                                        frameState.predictedDisplayTime);
 
+        //XrMath::Vector3f::
+        XrVector3f leftVec = {
```

Comment:
what's this doing?

Reply:
It calculates distance between head and hand. If hand is too close to head then it disables the immersive mode.